bbc/simorgh
# Build time variables namespaced

Resolves #1181 and #1168

There is currently a bug on simorgh TEST and LIVE where the client request (secondary request) gets data from the `window.location` which means that simorgh on TEST and LIVE is using fixture data. 

Since removing Razzle we lost the ability to have "build time variables" which causing many bugs when simorgh is on the client. These bugs were not seen due to simorgh on TEST and LIVE requesting static fixture data rather than getting it's data from Ares. 

This follows on from https://github.com/bbc/simorgh/pull/1151 as it requires both the `dotenv` logic and the application to be using Ares data on TEST and LIVE. This PR changes the data request url to be consistent across the client and server. It also creates "build time variables" that the client can access for any env var prefixed with `SIMORGH_`


Testing this PR:
- change the value of `SIMORGH_BASE_URL` in the `.env` file to be `https://www.test.bbc.com`
- `npm run build && npm run dev`
- visit `http://localhost:7080/news/articles/c5v767l2l0zo` if this page loads we can be certain that we are using Ares data because we do not have a local fixture for the article `c5v767l2l0zo`
- next visit another article (EG: http://localhost:7080/news/articles/c9rpqy7pmypo) and click one of the links
- Simorgh on the client will request the link with the new `SIMORGH_BASE_URL` we can check in the browser console where the data was fetched from, it should be `https://www.test.bbc.com/news/articles/$ARTICLE_ID.json` ~(the second request will be a broken page because of a CORS error\* which is currently being worked on by mozart and will be fixed in https://github.com/bbc/simorgh/issues/1012)~ work completed by Mozart
- Finally once this is built to TEST we should see the client request working without the CORS error which can be checked by going to `www.test.bbc.com/news/articles/c9rpqy7pmypo` and clicking the link through to the next article, and comparing the timestamp of the local fixture data and the Ares data

\* 
![image](https://user-images.githubusercontent.com/7791726/51253715-0370f000-1997-11e9-8077-c339e0f37191.png)
